PowerShell/JumpCloud Module/Public/Directory/Remove-JCOffice365Member.ps1

```diff
@@ -0,0 +1,106 @@
+function Remove-JCOffice365Member () {
+    param(
+        [Parameter( ValueFromPipelineByPropertyName, ParameterSetName = 'ByName', HelpMessage = 'The name of cloud directory instance')]
+        [String]$Name,
+        [Parameter( ValueFromPipelineByPropertyName, ParameterSetName = 'ByID', HelpMessage = 'The ID of cloud directory instance')]
+        [Alias('_id')]
+        [String]$ID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A username to remove to the directory')]
+        [String]$Username,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserID to remove to the directory')]
+        [String]$UserID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserGroup ID to remove to the directory')]
+        [String]$GroupID,
+        [Parameter( ValueFromPipelineByPropertyName, HelpMessage = 'A UserGroup name to remove to the directory')]
+        [String]$GroupName
+    )
+    begin {
+        Write-Debug 'Verifying JCAPI Key'
+        if ($JCAPIKEY.length -ne 40) {
+            Connect-JCOnline
+        }
+        $resultsArray = [System.Collections.Generic.List[PSObject]]::new()
+        $DirectoryHash = Get-JcSdkDirectory | Where-Object type -EQ 'office_365' | Select-Object id, name
+        if (($Username -or $UserID) -and ($GroupID -or $GroupName)) {
+            throw "Please use one type of association per call"
+        } elseif ($Username -and $UserID) {
+            throw "Please use either a username or a userID"
+        } elseif ($GroupID -and $GroupName) {
+            throw "Please use either a UserGroup Name or a UserGroup ID"
+        } elseif ($Username -or $UserID) {
+            Write-Debug 'Populating UserHash'
+            $UserHash = Get-DynamicHash -Object User -returnProperties username
+        } elseif ($GroupID -or $GroupName) {
+            Write-Debug 'Populating UserGroupHash'
+            $UserGroupHash = Get-DynamicHash -Object Group -GroupType User -returnProperties name
```

Comment:
Re: issues brought up with Add-JCGsuiteMember, I want to advocate for not building a hash for every user/ group